src/states/Preroll.js

```diff
@@ -247,12 +252,26 @@ class Preroll extends AdState {
     if (!player.ads._hasThereBeenALoadStartDuringPlayerLife) {
       videojs.log.warn('Leaving Preroll state before loadstart event can cause issues.');
     }
+    this.cleanupPartial(player);
+  }
 
+  /*
+   * Performs cleanup tasks without depending on a state transition. This is
+   * used mainly in cases where a preroll failed.
+   */
+  cleanupPartial(player) {
     player.removeClass('vjs-ad-loading');
     player.removeClass('vjs-ad-content-resuming');
-    player.clearTimeout(this._timeout);
+    this.clearTimeout(player);
   }
 
+  /*
+   * Clear the preroll timeout and nulls out the pointer.
+   */
+  clearTimeout(player) {
+    player.clearTimeout(this._timeout);
+    this._timeout = null;
```

Comment:
This mainly makes testing a little easier.

Reply:
Is this comment referring to setting `_timeout` to `null`? or this function as a whole? Would be good to add a comment to the function regarding it's usage for testing (if we want to keep it) as I could see this being refactored out easily.